### PR TITLE
[Workspace] Add Workspace filter to saved objects management page

### LINKED ISSUE #6098: [Workspace] Add workspace column and filter to saved objects page
**Is your feature request related to a problem? Please describe.**

When workspace feature turned on, saved objects will have a new attribute named workspace, to better filter and show which workspace they belong to, we are going to add a new workspace column in saved objects page.

A mockup page

image

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**
#4944

### PR DESCRIPTION
**Pull request by Hailong-am**
### Description

This PR follows up #6225 add workspace filter into saved objects page, so that user could filter saved objects by workspace.

### Issues Resolved

#6098

## Screenshot

https://github.com/opensearch-project/OpenSearch-Dashboards/assets/113890546/7a9ea0bd-a7fa-4157-b4d5-f2452ae21171

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/workspace/server/plugin.ts [hunk 1]
```diff
@@ -68,7 +68,7 @@ export class WorkspacePlugin implements Plugin<WorkspacePluginSetup, WorkspacePl
   }
 
   constructor(initializerContext: PluginInitializerContext) {
-    this.logger = initializerContext.logger.get('plugins', 'workspace');
+    this.logger = initializerContext.logger.get();
```
Comment (SuZhou-Joe) [inline src/plugins/workspace/server/plugin.ts:71]: Any reason that we remove the params here?

File: src/plugins/workspace/server/saved_objects/workspace_id_consumer_wrapper.ts [hunk 2]
```diff
@@ -63,8 +64,19 @@ export class WorkspaceIdConsumerWrapper {
           this.formatWorkspaceIdParams(wrapperOptions.request, options)
         ),
       delete: wrapperOptions.client.delete,
-      find: (options: SavedObjectsFindOptions) =>
-        wrapperOptions.client.find(this.formatWorkspaceIdParams(wrapperOptions.request, options)),
+      find: (options: SavedObjectsFindOptions) => {
```
Comment (SuZhou-Joe) [inline src/plugins/workspace/server/saved_objects/workspace_id_consumer_wrapper.ts:67]: The logic used to be in saved objects management page, any reason why we change to this wrapper?

File: src/plugins/workspace/server/plugin.ts (hunk 1 above)
Comment (Hailong-am) [reply src/plugins/workspace/server/plugin.ts:71]: the logger have the context already here https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/core/server/plugins/plugin_context.ts#L196

```typescript
/**
 * Plugin-scoped logger
 */
logger: {
  get(...contextParts) {
    return coreContext.logger.get('plugins', pluginManifest.id, ...contextParts);
  },
},
```

File: src/plugins/workspace/server/saved_objects/workspace_id_consumer_wrapper.ts (hunk 2 above)
Comment (Hailong-am) [reply src/plugins/workspace/server/saved_objects/workspace_id_consumer_wrapper.ts:67]: I need a central place to put this special logic, and this consumer seems to the place for workspace id related.

**Review by SuZhou-Joe — DISMISSED** [review]
LGTM

**Review by SuZhou-Joe — DISMISSED** [review]
LGTM

**Review by SuZhou-Joe — APPROVED** [review]

**Review by ZilongX — APPROVED** [review]

**Pull request merged**